### DISCUSSION TITLE
Regarding GSoC 2024

### DISCUSSION THREAD

**Discussion by ayushsinha25** [sosy-lab/benchexec#1001] [General]
Hi @PhilippWendler, 
One of the projects under SOSY that has interested me is under your mentorship ,i.e., Improvements to generated tables for Benchexec as it aligns with my skill set. How should I start contributing and is there something else I should know before contributing?

**Comment by PhilippWendler** [comment]
Hi and welcome!

I don't think there is anything particular that you need to know before contributing. For getting to know BenchExec I would in general recommend to first get started with understanding BenchExec, for example by reading our [paper](https://doi.org/10.1007/s10009-017-0469-y) and trying out BenchExec in practice.

For the generated tables, we have an [online example](https://sosy-lab.github.io/benchexec/example-table/svcomp-simple-cbmc-cpachecker.table.html). There are many improvement ideas about them, have a look at our issues labeled with [HTML table](https://github.com/sosy-lab/benchexec/issues?q=is%3Aopen+label%3A%22HTML+table%22).

If you have questions just ask me.

Greetings
Philipp

**Reply by ayushsinha25** [reply]
Ok, thanks a lot

**Reply by ayushsinha25** [reply]
> Hi and welcome!
> 
> 
> 
> I don't think there is anything particular that you need to know before contributing. For getting to know BenchExec I would in general recommend to first get started with understanding BenchExec, for example by reading our [paper](https://doi.org/10.1007/s10009-017-0469-y) and trying out BenchExec in practice.
> 
> 
> 
> For the generated tables, we have an [online example](https://sosy-lab.github.io/benchexec/example-table/svcomp-simple-cbmc-cpachecker.table.html). There are many improvement ideas about them, have a look at our issues labeled with [HTML table](https://github.com/sosy-lab/benchexec/issues?q=is%3Aopen+label%3A%22HTML+table%22).
> 
> 
> 
> If you have questions just ask me.
> 
> 
> 
> Greetings
> 
> Philipp

Hey,
I might need few help before starting.
-Am unable to set it up on my windows system
-What exactly we have to do in issue #924 , am little confused as there are no images for reference.
-Was #830 already resolved?

**Reply by PhilippWendler** [reply]
The benchmarking part of BenchExec doesn't work on Windows, because Windows doesn't provide the necessary features. If you want to try that out, you could use a VM with Linux. If you want to work only on the HTML tables, you in principle don't need it, though. But you should be able to run `table-generator` with some existing result files. `table-generator` only needs Python and works on Windows. There are some (old) existing result files in the directory `benchexec/tablegenerator/test_integration/` and [here](https://github.com/sosy-lab/benchexec/tree/gh-pages/example-table).

The filter overlay mentioned in #924 is the side bar that opens when you click the filter button in the top right corner. There are cases where you have textual filters, such as for the task, but depending on the table content also below.

#830 is not resolved yet, the link with the reproducer still shows it.

**Comment by ayushsinha25** [comment]
Ok, thanks a lot

**Comment by ayushsinha25** [comment]
Sir, I went through the codebase once, there are few questions-
What are the issues that you'd recommend to be taken up for GSoC?
Is there any proposal format for SOSY?

**Reply by PhilippWendler** [reply]
We have some issues labeled with [GSoC](https://github.com/sosy-lab/benchexec/issues?q=is%3Aissue+is%3Aopen+label%3AGSoC), these are likely good starting points.
We don't require a specific proposal format, if you follow Google's recommendations this is fine for us.
